helloquic crashes "INTERNAL_ERROR: no path to ..."

Hey Folks,

I've tried to run helloquic from the newest scion-apps version (latest commit on master) but the client always crashes with "INTERNAL_ERROR: no path to". `scion showpaths` in contrast shows all paths and does not report any errors.

Tested on:
- Ubuntu 20.04
- Fresh scionlab user-AS installation from SCIONLab.org
- Go 1.17.7

System output:
```sh
root@ubuntu-4gb-nbg1-1-sc3:~/scion-apps/_examples/helloquic# ./helloquic -remote "19-ffaa:1:c3f,[127.0.0.1]:43000"
Fatal error: INTERNAL_ERROR: no path to 19-ffaa:1:c3f
root@ubuntu-4gb-nbg1-1-sc3:~/scion-apps/_examples/helloquic# scion showpaths 19-ffaa:1:c3f -r
Available paths to 19-ffaa:1:c3f
3 Hops:
[0] Hops: [19-ffaa:1:c59 1>127 19-ffaa:0:1303 121>1 19-ffaa:1:c3f] MTU: 1472 NextHop: 127.0.0.1:30001 Status: alive LocalIP: 127.0.0.1
4 Hops:
[1] Hops: [19-ffaa:1:c59 1>127 19-ffaa:0:1303 247>1 19-ffaa:1:cf0 11>11 19-ffaa:1:c3f] MTU: 1472 NextHop: 127.0.0.1:30001 Status: alive LocalIP: 127.0.0.1
[2] Hops: [19-ffaa:1:c59 1>127 19-ffaa:0:1303 247>1 19-ffaa:1:cf0 12>12 19-ffaa:1:c3f] MTU: 1472 NextHop: 127.0.0.1:30001 Status: alive LocalIP: 127.0.0.1
[3] Hops: [19-ffaa:1:c59 1>127 19-ffaa:0:1303 247>1 19-ffaa:1:cf0 13>13 19-ffaa:1:c3f] MTU: 1472 NextHop: 127.0.0.1:30001 Status: alive LocalIP: 127.0.0.1
[4] Hops: [19-ffaa:1:c59 1>127 19-ffaa:0:1303 247>1 19-ffaa:1:cf0 14>14 19-ffaa:1:c3f] MTU: 1472 NextHop: 127.0.0.1:30001 Status: alive LocalIP: 127.0.0.1
```
Note: Running the helloquic server in the same AS does not produce this error, but it crashes with:
```
gotcha: hi dude, 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x9e3db1]

goroutine 1 [running]:
github.com/netsec-ethz/scion-apps/pkg/pan/internal/ping.(*Pinger).Close(...)
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/pkg/pan/internal/ping/ping.go:141
github.com/netsec-ethz/scion-apps/pkg/pan.(*PingingSelector).Close(0xc0000eadc0)
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/pkg/pan/selector.go:311 +0x31
github.com/netsec-ethz/scion-apps/pkg/pan.(*dialedConn).Close(0xc0002a25b0)
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/pkg/pan/udp_dial.go:153 +0x66
github.com/netsec-ethz/scion-apps/pkg/pan.(*QUICSession).CloseWithError(0xc0000ad040, 0xc00000e018, {0x0, 0x3})
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/pkg/pan/quic_dial.go:35 +0x4e
main.runClient({0x7ffd6eab7e95, 0x1f}, 0x1)
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/_examples/helloquic/helloquic.go:142 +0x4cc
main.main()
        /home/marten/go/src/github.com/netsec-ethz/scion-apps/_examples/helloquic/helloquic.go:51 +0x193
```
PS: I spent the last days debugging some QUIC connection issues, in case I've overlooked something obvious, just ignore the first part :)

--- a/pkg/pan/selector.go
+++ b/pkg/pan/selector.go
@@ -159,6 +159,7 @@ func (s *PingingSelector) Initialize(local, remote UDPAddr, paths []*Path) {
 
 	s.local = local.scionAddr()
 	s.remote = remote.scionAddr()
+	s.paths = paths
 	s.current = stats.LowestLatency(s.remote, s.paths)
 }
 
@@ -184,6 +185,10 @@ func (s *PingingSelector) reselectPath() {
 func (s *PingingSelector) ensureRunning() {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
+
+	if s.local.IA == s.remote.IA {
+		return
+	}
 	if s.pinger != nil {
 		return
 	}
@@ -211,7 +216,7 @@ func (s *PingingSelector) run() {
 	for {
 		select {
 		case <-s.pingerCtx.Done():
-			break
+			return
 		case <-pingTicker.C:
 			numActive := int(atomic.LoadInt64(&s.numActive))
 			if numActive > len(s.paths) {
@@ -307,6 +312,11 @@ func (s *PingingSelector) handlePingReply(reply ping.Reply,
 }
 
 func (s *PingingSelector) Close() error {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	if s.pinger == nil {
+		return nil
+	}
 	s.pingerCancel()
 	return s.pinger.Close()
 }

